### ISSUE TITLE
Truncation behaviour for internal labels in non-interactive components

### ISSUE THREAD

**Issue by fbd399** [mi6/ic-ui-kit#332]
## Summary
Define how to handle internal labels which are longer than the non-interactive components they are housed in, like Chips, and if applicable, the interaction to view the full label.

## 💰 User value
Accessibility importance stems from multi-channel users needing to view full label whilst still maintaining the correct formatting of the components.

## 📚 User Stories

*As an* ICDS user
*I need* to see the full content of internal labels
*So that* I have all the necessary information

## 📝 Acceptance Criteria 

**Scenario 1: Chip/Status tag wraps content when internal label container is wider than component container**

Given a Chip/Status tag component with an internal label container width that is wider than the component container,
When the content of the Chip/Status tag is rendered,
Then the content should wrap to fit within the component container.

**Scenario 2: Maintain fixed border radius values for Chip/Status tag component when wrapping**

Given a Chip/Status tag component with wrapped content,
When the border radius values are applied,
Then the border radius should be maintained at the fixed values as follows:

Small: 14px
Default: 16px
Large: 20px

**Scenario 3: Content and icon are vertically aligned to top when text wraps**

Given a Chip/Status tag component with wrapped content,
When the content and icon are displayed within the component,
Then the content and icon should be vertically aligned to the top of the component.

## ✏ Designs
Component Contribution file: Truncation (internal labels) - Component Contribution

## 🧾 Guidance
Content and imagery updated in relevant files.

**Comment by fbd399** [comment]
Updated ticket to reflect design decisions. Refer to the Contribution files for specific design comments. Decisions reflect dev feedback. @MI6-451 are you able to note down the technical considerations we discussed?

**Comment by MI6-451** [comment]
Rather than add any truncation behaviour we would be adding a max character length to chips, either a strict max length which would cut off any additional text or a console error to let users know they have exceeded the set max length. This max length would need to be clearly stated in the guidance.

**Comment by DBD324** [comment]
Need to avoid using the console log to show this error. Instead, we will include guidance and detail to the prop description that indicates the character limit. The components will allow entries larger than the character limit, but this ticket needs to define how those components will display it. 
Options are:
1. Truncate - a11y issue with hiding information that can't be obtained without hover/focus state. 
2. Wrap - The ideal solution but would need to assess impact when multiple elements of different content sizes are placed in situ. 
3. Cut off - Not recommended as the content is lost and no affordance is provided that content has been cut off.

**Comment by fbd399** [comment]
Considering all available options we’ll avoid console error handling, and update Chip/Status tag designs to more neatly wrap multi-line labels. Namely, to change border radius from a percentage value to a fixed value so that border radius is maintained when wrapping.

Change border radius from percentage to fixed amount:
Small: 14px
Default: 16px
Large: 20px

Make sure that if text wraps, the content is vertically aligned to top (refer to Outlines.png).

Guidance informs users to use clear and concise labels to avoid wrapping. Updated guidance to reflect this. Updated imagery in relevant design files for Status tags, and new imagery for Chips.


**Actions:
- Change border radius to fixed number for Status tags and Chips
- Update the guidance text for both Status tags and Chips
- Upload imagery to guidance site for Status tags and Chips**



![Image](https://user-images.githubusercontent.com/117087335/228307687-09401623-3ef6-4cef-ab14-7d0a7817c644.png)
![Image](https://user-images.githubusercontent.com/117087335/228307688-9ec7ba1e-c216-41a5-98ca-8fb3e615a8a6.png)
